src/core/cgroup.c

```diff
@@ -1200,21 +1206,46 @@ static int cgroup_apply_devices(Unit *u) {
         return r;
 }
 
-static void set_io_weight(Unit *u, const char *controller, uint64_t weight) {
-        char buf[8+DECIMAL_STR_MAX(uint64_t)+1];
-        const char *p;
+/* Convert the normal io.weight value to io.bfq.weight */
+#define BFQ_WEIGHT(weight) \
```

Comment:
i'd really prefer if this was a function, i.e. this macro doesn't enclose argumnts in () as one should, and doesn't guard against double eval, and falls totally apart if negative values/incorrect types are passed. And the ability of being able to use compile time verification instead of test runtime doesn't really matter I think.

Reply:
FTR, that's been cleaned in https://github.com/systemd/systemd/pull/21728/commits/d2bd0bfa362a26a8cd6e1d68e5c485f129fbb379. But I'll take into account your remark when getting back to https://github.com/systemd/systemd/pull/21795/commits/d473255080951df2d3710e010b21896e8774e63f (dropping it).